i3_resurrect/programs.py

```diff
@@ -17,7 +17,10 @@ def save(workspace, numeric, directory, profile):
     Save the commands to launch the programs open in the specified workspace
     to a file.
     """
-    filename = f'workspace_{workspace}_programs.json'
+    # Filter title string for saving in filename
+    workspace_id = util.filename_filter(workspace)
+
+    filename = f'workspace_{workspace_id}_programs.json'
```

Comment:
Here you leave a comment and blank line, but below (line 86) you do not. Might be good to be more consistent.

Reply:
Removed comment here. filename_filter function already have one.